src/redux/reducers.js

```diff
@@ -1276,3 +1276,83 @@ export function usernameSubscribers(state = {}, action) {
 export function usernameSubscriptions(state = {}, action) {
   return handleSubs(state, action, ActionTypes.SUBSCRIPTIONS)
 }
+
+const removeItemFromGroupRequests = (state, action) => {
+  const userName = action.request.userName
+  const groupName = action.request.groupName
+  
+  const group = state.find(group => group.username === groupName)
+
+  if (group && group.requests.length !== 0) {
+    let newGroup = {
+      ...group, 
+      requests: group.requests.filter(user => user.username !== userName)
+    }
+
+    return _(state).without(group).push(newGroup).value()
+  }
+
+  return state
+}
+
+export function groupRequests(state = [], action) {
+  switch (action.type) {
+    case response(ActionTypes.GROUP_REQUESTS): {
+      return action.payload.map(group => {
+        group.requests = group.requests.map(userParser)
+        return {...group}
+      })
+    }
+    case response(ActionTypes.ACCEPT_GROUP_REQUEST):
+    case response(ActionTypes.REJECT_GROUP_REQUEST): {
+      return removeItemFromGroupRequests(state, action)
+    }
+  }
+
+  return state
+}
+
+export function requests(state = [], action) {
+  switch (action.type) {
+    case response(ActionTypes.WHO_AM_I): {
+      return action.payload.requests.map(userParser)
+    }
+    case response(ActionTypes.ACCEPT_USER_REQUEST):
+    case response(ActionTypes.REJECT_USER_REQUEST): {
+      const userName = action.request.userName
+      return state.filter((user) => user.username !== userName)
+    }
+  }
+
+  return state
+}
+
+export function groupRequestsCount(state = 0, action) {
+  switch (action.type) {
+    case response(ActionTypes.GROUP_REQUESTS): {
+      return action.payload.reduce((acc, group) => {
+        return acc + group.requests.length
+      }, 0)
+    }
+    case response(ActionTypes.ACCEPT_GROUP_REQUEST):
+    case response(ActionTypes.REJECT_GROUP_REQUEST): {
+      return Math.max(0, state - 1)
+    }
+  }
+
+  return state
+}
+
+export function requestsCount(state = 0, action) {
```

Comment:
OK, so it's fine to have two separate reducers. But maybe it'd be good to rename this one to `userRequestsCount`? WDYT?